gprofiler/merge.py

```diff
@@ -250,9 +252,11 @@ def concatenate_profiles(
 
     for pid, stacks in process_profiles.items():
         container_name = _get_container_name(pid, docker_client, add_container_names)
+        application_name = get_application_name(pid) or "" if add_application_names else ""
+        prefix = container_name + ";" if add_container_names else "" + application_name
```

Comment:
We'll be missing `;` for the `application_name`, because here you don't seem to replace the first frame (which is what I recall we have decided earlier). I think we should just replace the first frame name (i.e `python` `celery` whatever into your new app name)

Reply:
Not relevant anymore..